build-support/bin/ci-failure.sh

```diff
@@ -1,7 +1,7 @@
 #!/bin/bash -eu
 
 if [[ "$(uname)" == "Darwin" ]]; then
-  if (( $(ls -1 /cores | wc -l) > 0 )); then
+  if (( $(find /cores -maxdepth 1 ! -name "$(basename /cores)" | wc -l) > 0 )); then
```

Comment:
Teach me! This seems worse than what was there before... What's wrong with the old way?

Reply:
Agreed that it's more verbose. This change is because you should never use `ls` in scripts. `ls` is only designed for humans and not machines, e.g. its output is not stable. See https://mywiki.wooledge.org/ParsingLs.

The specific issue this fixes is https://github.com/koalaman/shellcheck/wiki/SC2012.

Reply:
According to https://mywiki.wooledge.org/ParsingLs the thing you're guarding against is that filenames may contain newlines, and these two approaches have the same problem there... I won't block the PR, but I don't like this particular change...